Missing C++ Build Flag causing pip failure on MacOS Mojave Python 3.7

Thanks for the great project! I have been using your library with some success in the past, but noticed an issue on a fresh install that I was hoping I could help out with.

I just tried to do a pip install into a fresh python 3.7 virtual environment on a MacOS Mojave system and received the following error:

```
pip install ghalton
Collecting ghalton
  Downloading https://files.pythonhosted.org/packages/36/d3/88ccf26adde4930247dd58367ba896d149579fcbbe381e278e7c9b25db09/ghalton-0.6.1.tar.gz (125kB)
     |████████████████████████████████| 133kB 786kB/s
Building wheels for collected packages: ghalton
  Building wheel for ghalton (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: [directory obfuscated]/miniconda3/envs/samply/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/setup.py'"'"'; __file__='"'"'/private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-wheel-c4h3ob5i --python-tag cp37
       cwd: /private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/
  Complete output (20 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.7-x86_64-3.7
  creating build/lib.macosx-10.7-x86_64-3.7/ghalton
  copying ghalton/ghalton_wrapper.py -> build/lib.macosx-10.7-x86_64-3.7/ghalton
  copying ghalton/constants.py -> build/lib.macosx-10.7-x86_64-3.7/ghalton
  copying ghalton/__init__.py -> build/lib.macosx-10.7-x86_64-3.7/ghalton
  running build_ext
  building 'ghalton._ghalton_wrapper' extension
  creating build/temp.macosx-10.7-x86_64-3.7
  creating build/temp.macosx-10.7-x86_64-3.7/src
  gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I[directory obfuscated]/miniconda3/envs/samply/include -arch x86_64 -I[directory obfuscated]/miniconda3/envs/samply/include -arch x86_64 -I[directory obfuscated]/miniconda3/envs/samply/include/python3.7m -c src/Halton_wrap.cxx -o build/temp.macosx-10.7-x86_64-3.7/src/Halton_wrap.o
  warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
  src/Halton_wrap.cxx:3044:10: fatal error: 'stdexcept' file not found
  #include <stdexcept>
           ^~~~~~~~~~~
  1 warning and 1 error generated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for ghalton
  Running setup.py clean for ghalton
Failed to build ghalton
Installing collected packages: ghalton
    Running setup.py install for ghalton ... error
    ERROR: Command errored out with exit status 1:
     command: [directory obfuscated]/miniconda3/envs/samply/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/setup.py'"'"'; __file__='"'"'/private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-record-d_u0q2gl/install-record.txt --single-version-externally-managed --compile
         cwd: /private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/
    Complete output (20 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.7-x86_64-3.7
    creating build/lib.macosx-10.7-x86_64-3.7/ghalton
    copying ghalton/ghalton_wrapper.py -> build/lib.macosx-10.7-x86_64-3.7/ghalton
    copying ghalton/constants.py -> build/lib.macosx-10.7-x86_64-3.7/ghalton
    copying ghalton/__init__.py -> build/lib.macosx-10.7-x86_64-3.7/ghalton
    running build_ext
    building 'ghalton._ghalton_wrapper' extension
    creating build/temp.macosx-10.7-x86_64-3.7
    creating build/temp.macosx-10.7-x86_64-3.7/src
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I[directory obfuscated]/miniconda3/envs/samply/include -arch x86_64 -I[directory obfuscated]/miniconda3/envs/samply/include -arch x86_64 -I[directory obfuscated]/miniconda3/envs/samply/include/python3.7m -c src/Halton_wrap.cxx -o build/temp.macosx-10.7-x86_64-3.7/src/Halton_wrap.o
    warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
    src/Halton_wrap.cxx:3044:10: fatal error: 'stdexcept' file not found
    #include <stdexcept>
             ^~~~~~~~~~~
    1 warning and 1 error generated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: [directory obfuscated]/miniconda3/envs/samply/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/setup.py'"'"'; __file__='"'"'/private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-install-lkc_y4n8/ghalton/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/sc/j5wn6cv55pj8d65qg4rgdn715k4qkm/T/pip-record-d_u0q2gl/install-record.txt --single-version-externally-managed --compile Check the logs for full command output.
```

I feel like I have dealt with this problem before on my own projects, and I was able to fix it by doing what the compiler told me:

```
    warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
```

I don't know if this will affect builds on other systems, so I will put together a PR for you to review, if you don't mind.

--- a/setup.py
+++ b/setup.py
@@ -1,9 +1,12 @@
 #!/usr/bin/env python
 from setuptools import setup, find_packages, Extension
 
-ghalton_module = Extension("ghalton._ghalton_wrapper", sources=["src/Halton_wrap.cxx", "src/Halton.cpp"])
+ghalton_module = Extension("ghalton._ghalton_wrapper",
+                           sources=["src/Halton_wrap.cxx", "src/Halton.cpp"],
+                           extra_compile_args=["-stdlib=libc++"],
+                           extra_link_args=['-stdlib=libc++'])
 
-version = "0.6.1"
+version = "0.6.2"
 
 setup (name = "ghalton",
        version = version,

